core/src/main/java/org/infinispan/persistence/file/SingleFileStore.java

```diff
@@ -765,21 +834,24 @@ public SingleFileStoreConfiguration getConfiguration() {
        */
       private transient int readers = 0;
 
+      private transient MarshalledEntry marshalledEntry;
+
       public FileEntry(long offset, int size) {
-         this(offset, size, 0, 0, 0, -1);
+         this(offset, size, 0, 0, 0, -1, null);
       }
 
-      public FileEntry(long offset, int size, int keyLen, int dataLen, int metadataLen, long expiryTime) {
+      public FileEntry(long offset, int size, int keyLen, int dataLen, int metadataLen, long expiryTime, MarshalledEntry marshalledEntry) {
          this.offset = offset;
          this.size = size;
          this.keyLen = keyLen;
          this.dataLen = dataLen;
          this.metadataLen = metadataLen;
          this.expiryTime = expiryTime;
+         this.marshalledEntry = marshalledEntry;
```

Comment:
Isn't this going to keep all serialized values in memory forever?

`FileEntry` should be as small as possible, because we need to keep a `FileEntry` instance in the heap for each entry in the store.